chainermn/communicators/_base.py

```diff
@@ -15,6 +20,57 @@ def rank(self):
     def size(self):
         return self.mpi_comm.size
 
+    def send(self, array, dest, tag):
+        """A primitive feature of inter-process transmission.
+
+        This method sends numpy-array to target process.
+        This method relies on mpi4py fast communication optimized for
+        numpy arrays, which discards any information attached to
+        chainer.Variable objects. Please be sure.
+
+        Args:
+            array: numpy or cupy array object.
+            dest (int): Target process specifier.
+            tag (int): An optional message ID.
+
+        """
+        chainer.utils.experimental(
+            'chainermn.communicators.CommunicatorBase.send')
+        assert array.dtype == numpy.float32
+        ndim = numpy.array([array.ndim], dtype=numpy.int32)
+        shape = numpy.array(array.shape, dtype=numpy.int32)
+        buf = _memory_utility.array_to_buffer_object(array)
+        self.mpi_comm.Send([ndim, mpi4py.MPI.INT], dest=dest, tag=tag)
+        self.mpi_comm.Send([shape, mpi4py.MPI.INT], dest=dest, tag=tag)
+
+        if chainer.cuda.get_array_module(array) is not numpy:
+            chainer.cuda.Stream.null.synchronize()
+        self.mpi_comm.Send(buf, dest=dest, tag=tag)
+
+    def recv(self, source, tag):
+        """A primitive feature of inter-process transmission.
```

Comment:
The summary lines for `send` and `recv` is the same, could you make it a little more descriptive?